R/app.R

```diff
@@ -288,18 +295,37 @@ initAutoReloadMonitor <- function(dir) {
   obs$destroy
 }
 
+# Loads in all helpers in the R/ directory of the app
+# From `list.files`:
+# > The files are sorted in alphabetical order, on the full path
+loadHelpers <- function(appDir, envir=globalenv()){
+  helpersDir <- file.path(appDir, "R")
+  if (!file.exists(helpersDir)){
+    # We may be on a case-sensitive filesystem and have a dir named r/
+    helpersDir <- file.path(appDir, "r")
+  }
+  helpers <- list.files(helpersDir, pattern="\\.[rR]$", recursive=TRUE, full.names=TRUE)
```

Comment:
I'd also suggest not loading recursively, again to increase compatibility with package code loading.

Reply:
Ah, I didn't realize packages were non-recursive.

Do you feel strongly on this point? I feel like large codebases will require multiple directories to organize content (the Angular structure comes to mind [e.g.](https://scotch.io/tutorials/angularjs-best-practices-directory-structure)). 

You could obviously accomplish that with R files at the base that `source` the inner directories, but that feels odd to me. Do people often have R files nested inside directories within `R/` that would be problematic if they were sourced?

Reply:
Oh, counter-point to my previous comment: if we want to maintain compatibility with golem (i.e. allow users to painlessly flip this structure over to using an R package as the organizational construct), then supporting something that R packages don't support would be problematic.

Maybe we just do the top-level for now and then if/when packages support nested directories, we can follow suit.

Reply:
:+1: